### PR TITLE
#2553 fix: http server command case sensitivity

### PR DESCRIPTION
**Pull request by robfrank**
This pull request improves the robustness of server command handling by ensuring that all supported commands are parsed and executed correctly regardless of their case (uppercase, lowercase, or mixed case). It also introduces comprehensive regression and parameterized tests to verify this behavior for a wide range of commands, including database operations, user management, server settings, and cluster management.

### Command Parsing and Execution Improvements

* Updated `PostServerCommandHandler.java` so that all command parsing and target extraction use the lowercased version of the command (`command_lc`) instead of the original input, fixing case sensitivity issues for all supported commands.

### Comprehensive Case Sensitivity Testing

* Added multiple parameterized and regression tests to `PostServerCommandHandlerIT.java` that verify database operations, user management, server settings, and cluster management commands work correctly with uppercase, lowercase, mixed case, and extra whitespace.
* Added specific regression tests for issue #2553 to ensure database operations do not fail due to command case mismatches.
* Introduced edge case tests for malformed commands and confirmed that appropriate error codes are returned, documenting actual behavior.
* Added backward compatibility tests to ensure that existing command patterns and functionality are not broken by the case sensitivity fix.

### Test Infrastructure Enhancements

* Added helper methods and imported JUnit parameterized test utilities to support efficient and consistent testing of command variations.

### CODE REVIEW

File: server/src/main/java/com/arcadedb/server/http/handler/PostServerCommandHandler.java [hunk 1]
```diff
@@ -88,32 +88,32 @@ public ExecutionResponse execute(final HttpServerExchange exchange, final Server
...
+      setDatabaseSetting(extractTarget(command_lc, SET_DATABASE_SETTING));
     else if (command_lc.startsWith(SET_SERVER_SETTING))
-      setServerSetting(extractTarget(command, SET_SERVER_SETTING));
+      setServerSetting(extractTarget(command_lc, SET_SERVER_SETTING));
     else if (command_lc.startsWith(GET_SERVER_EVENTS))
-      response.put("result", getServerEvents(extractTarget(command, GET_SERVER_EVENTS)));
+      response.put("result", getServerEvents(extractTarget(command_lc, GET_SERVER_EVENTS)));
     else if (command_lc.startsWith(ALIGN_DATABASE))
-      alignDatabase(extractTarget(command, ALIGN_DATABASE));
+      alignDatabase(extractTarget(command_lc, ALIGN_DATABASE));
```
[the remark this answers is not in the document]
Comment (gramian) [reply server/src/main/java/com/arcadedb/server/http/handler/PostServerCommandHandler.java]: See https://github.com/ArcadeData/arcadedb/pull/2556
